contentgrid-spring-data-rest/src/main/java/com/contentgrid/spring/data/rest/webmvc/DefaultDomainTypeToHalFormsPayloadMetadataConverter.java

```diff
@@ -227,4 +244,34 @@ public Class<?> getType() {
         }
     }
 
+    @FunctionalInterface
+    static interface PropertyMetadataFactory {
+        PropertyMetadata build(Property property, Class<?> domainClass, String path);
+    }
+
+    private static String underscoreToCamelCase(String underscored) {
```

Comment:
Does this also work for attributes with a reserved java keyword name? They are prefixed with an underscore in scribe. e.g. `_public` should not be changed into `Public`.

Reply:
Those would still have `@JsonProperty("public")`, right?